content/blogposts/2023/blah.md

```diff
@@ -0,0 +1,138 @@
+tags: [blog]
+slug: '(TODO - update slug)'
+published: true
+heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/(TODO - get hero/social image).png
+socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/(TODO - get hero/social image).png
+---
+
+(TODO - attention grabber and introduction)
+
+## Orthogonal search dimensions
+
+(TODO - rewrite this paragraph) Consider embeddings-powered search to be good at answering high-level questions. You're asking a question about auth, of course we want to include relevant files about auth docs, etc. Ask a question about how code is implemented, suddenly the 10,000 foot view doesn't make sense. Using our heap of compiler-accurate source code indexes (SCIP), we can traverse semantic relationships between pieces of code and include the relevant ones in the context window. This gives us a detail-oriented source view on the same data.
+
+<a href="https://storage.googleapis.com/sourcegraph-assets/blog/scip-powered-cody/architecture.png" target="_blank">
+    <Figure
+        src="https://storage.googleapis.com/sourcegraph-assets/blog/scip-powered-cody/architecture.png"
+        alt="An sequence diagram showing the steps of resolving precise code intelligence context for a Cody query." />
+</a>
+
+(TODO - rewrite this paragraph) The context sent to the API is anything that the user can reasonably see: the current repository, the current commit, the closest commit on remote, a list of open files, the user’s text selection, a diff between the editor state and remote; or a particular user query or recipe invocation. From here, the code intelligence context service can perform a cheap syntactic analysis (via TreeSitter) to extract symbol names from the currently visible code. These symbol names can be translated into SCIP names, giving us a set of root symbols to begin a traversal of relationships. The API gathers relevant code metadata, locations, and source text, and returns it to the Cody client for inclusion in the context window. Following different types of relationship edges will have different effects based on the user’s intent. If a user is looking to explain or generate a unit test for an arbitrary piece of code, it’s intuitive that exposing the implementation of invoked functions will be beneficial. If a user is generating a user form that populates a given type, it’s similarly intuitive that exposing the definitions and documentation of each of the type’s properties would prevent a response with hallucinated or mis-typed form fields. Supposing that the Sourcegraph Enterprise has also indexed package hosts, the global SCIP index will also include all relevant downstream dependencies. This allows us to traverse very deeply into the symbol graph to gather relevant context.
+
+(TODO - introduce the following examples section)
+
+### Risperidone
+
+**As it turns out, the greatest LLM anti-psychotic is giving it enough context in the first place.**
```

Comment:
Nitpick at this stage, but I think "enough of _the right context_" is the thing here :)